gpumon/gpumon_server.ml

```diff
@@ -70,12 +70,17 @@ module Make(I: Interface):IMPLEMENTATION = struct
       | Gpumon_interface.(Gpumon_error (NvmlFailure _)) as err -> raise err
       | err -> raise Gpumon_interface.(Gpumon_error (NvmlFailure (Printexc.to_string err)))
 
-    let get_vgpu_metadata _dbg domid pgpu_address =
+    let get_vgpu_metadata _dbg domid pgpu_address vgpu_uuid =
       let interface = get_interface_exn () in
       let domid'    = string_of_int domid  in
+      let filter_instances = match vgpu_uuid with
+           | "" -> List.map (fun vgpu -> vgpu)
```

Comment:
This creates a special case for when the uuid is the empty string. Would it not be better to make `vgpu_uuid` and option type to make this more explicit?

Reply:
Good idea, let me figure it out.

Reply:
Changing the parameter optional requires changing in IDL. But we didn't figure out a way to change. Agreed with Lindig to leave it for future improvement.